src/neo4clj/cypher.clj

```diff
@@ -64,18 +66,37 @@
        (when props (properties props))
        "]->" to ))
 
-(defn lookup
-  "Takes a lookup representation and returns its cypher equvalent
+(defn lookup-where
+  [{:keys [ref-id id props]}]
+  (cond
+    id (str "ID(" ref-id ") = " id)
+    props (where ref-id props)
+    :else nil))
+
+(defn lookup-node
+  "Takes a node lookup representation and returns its cypher equvalent
 
   The return value is an vector with the first part being the actual entity and
   the second is the where clause for the lookup."
-  [{:keys [id ref-id props] :as lookup}]
+  [{:keys [ref-id id props] :as lookup}]
   (vector (str "(" ref-id (labels (:labels lookup))
                (when (map? props) (properties props))
                ")")
-          (when-let [where-parts (not-empty
-                                  (remove nil? [(when id
-                                                  (str "ID(" ref-id ") = " id))
-                                                (when (and props (not (map? props)))
-                                                  (str "(" (where ref-id props) ")"))]))]
-            (str/join " AND " where-parts))))
+          (when (or id (not (map? props)))
+            (lookup-where lookup))))
+
+(defn lookup-relationship
+  "Takes a relationship lookup representation and returns its cypher equvalent
+
+  The return value is an vector with the first part being the actual entity and
+  the second is the where clause for the lookup."
+  [{:keys [ref-id id type from to props] :as rel}]
+  (let [[from-cypher from-cypher-where] (lookup-node from)
+        [to-cypher to-cypher-where] (lookup-node to)]
+    (vector (relationship from-cypher to-cypher rel)
+            (when-let [where-parts (not-empty (remove nil? (vector
```

Comment:
I think threading macros could ease readability:
```clojure
; yes - I already removed the `when-let` part :grin: 
(->> (vector (lookup-where rel) from-cypher-where to-cypher-where) ; I also removed `when`
     (remove nil?)
     ;;(not-empty) ; <- This isn't needed anymore
     (str/join " AND "))

Reply:
Your change results in this function not working as the rest of the lookup functions. All lookup functions today returns a nil if no WHERE parts is present, with this change it will be an empty string.

Also the `(when (or id (not (map? props)))` is there for the same purpose as stated above.

Reply:
If that is important, put `(not-empty)` AFTER `(str/join " AND ")`. and it should return `nil` again:
```clojure
; yes - I already removed the `when-let` part :grin: 
(->> (vector (lookup-where rel) from-cypher-where to-cypher-where) ; I also removed `when`
     (remove nil?)
     (str/join " AND ")
     (not-empty))
```